### PR TITLE
[DNM] enable set tiflash_mpp engine label using config

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:
1. Enable config **engine-label** for tiflash_proxy, check etc/config-template.toml 
2. Related tiflash_proxy change is in this [branch](https://github.com/guo-shaoge/tikv/tree/cloud_test).
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -261,7 +261,7 @@ struct TiFlashProxyConfig
                 args_map[engine_store_address] = config.getString("flash.service_addr");
             else
                 args_map[engine_store_advertise_address] = args_map[engine_store_address];
-            args_map[engine_label] = engine_label_value;
+            // args_map[engine_label] = engine_label_value;
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:264]: if there is no "flash.proxy.engine-label", then set the value default to be "tiflash" to keep the same behavior for write node.

File: dbms/src/Server/Server.cpp [hunk 2]
```diff
@@ -261,7 +261,10 @@ struct TiFlashProxyConfig
                 args_map[engine_store_address] = config.getString("flash.service_addr");
             else
                 args_map[engine_store_advertise_address] = args_map[engine_store_address];
-            args_map[engine_label] = engine_label_value;
+            if (args_map.find(engine_label) == args_map.end())
+            {
+                args_map[engine_label] = default_engine_label_value;
```
Comment (guo-shaoge) [inline dbms/src/Server/Server.cpp:266]: Give error when label is neither "tiflash" or "tiflash_mpp"

Comment (guo-shaoge) [inline dbms/src/Server/Server.cpp:266]: Give error when label is neither "tiflash" or "tiflash_mpp"

**Comment by JaySon-Huang** [comment]
Create a branch on the pingcap/tiflash repo as a feature branch, so that we can file further base on that branch https://github.com/pingcap/tiflash/pull/4805. Closing this one

**Pull request closed without merging**